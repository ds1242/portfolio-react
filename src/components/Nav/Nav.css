/* *{
    margin:0;
    padding:0;
    box-sizing: border-box;
  } */
/* :root{
    --primary_color:#0b132b;
    --secondary_color:#0DAB76;
    --tertiary_color:#edf2f4;
    --quartiary_color:#45503b;
    --cinqieme_color:#B1740F;
} */
/* body{
    font-family: 'Gill Sans', 'Gill Sans MT', Calibri, 'Trebuchet MS', sans-serif;
    color:var(--primary_color);
    background-color:#8d99ae; */
/* } */
nav {
    background-color:var(--primary_color);
}
nav a {
    color:var(--secondary_color); 
    text-shadow: 0 0 10px rgba(0,0,0,0.5);
}
nav h1:hover{
    color: var(--tertiary_color);
    text-shadow:2px 2px 5px var(--tertiary_color);
}
nav h3 {
    color:var(--secondary_color); 
    text-shadow: 0 0 10px rgba(0,0,0,0.5)
}
nav h3:hover{
    color: var(--tertiary_color);
    text-shadow:2px 2px 5px var(--tertiary_color);
}

/* @media (min-width: 576px) { ... } */

/* Medium devices (tablets, 768px and up) */
/* @media (min-width: 768px) { 
    
 } */

/* Large devices (desktops, 992px and up) */
/* @media (min-width: 992px) { ... }

/* Extra large devices (large desktops, 1200px and up) */
/* @media (min-width: 1200px) { ... } */ 
@media screen and (max-width:980px){
    header{
        padding-bottom: 0;
        justify-content: center;
        position:relative;
        width:100%;
        /* padding:0px; */
        /* margin:0; */
    }
    header h1{
        width:100%;
        text-align: center;
        font-size: 50px;
    }
    header nav {
        width: 100%;
    }
    header nav ul {
        margin-top: 20px;
        width:100%;
        justify-content: center;
    }
    header nav ul li a{
        font-size: 35px;
    }
}