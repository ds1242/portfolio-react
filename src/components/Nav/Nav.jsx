import React, { useState } from 'react';
import './Nav.css';



function Nav(props) {
    const {
        aboutSelected,
        setAboutSelected,
        workSelected,
        setWorkSelected,
        contactSelected,
        setContactSelected,
        resumeSelected,
        setResumeSelected
    } = props
    return (
        <header>            
        <nav className="navbar navbar-expand-sm" data-testid="navbar">
            <div className="container-fluid ">
                <a className="navbar-brand"  data-testid="link" href="/">
                    <h1>David's Portfolio</h1>
                </a>
                    <ul className="navbar-nav ">
                        <li>
                            <a className="nav-link active" data-testid="about" href="#about"                             
                            onClick={() => { setAboutSelected(true); setContactSelected(false); setResumeSelected(false); setWorkSelected(false); }}>
                            <h3>About Me</h3></a>
                        </li>
                        <li>
                            <a className="nav-link active" href="#work"
                            onClick={() => { setAboutSelected(false); setContactSelected(false); setWorkSelected(true); setResumeSelected(false); }}>
                            <h3>My Projects</h3></a>                            
                        </li>
                        <li>
                            <a className="nav-link active" href="#contact-me" 
                            onClick={() => { setAboutSelected(false); setContactSelected(true); setWorkSelected(false); setResumeSelected(false); }}>
                            <h3>Contact</h3></a>
                        </li>
                        <li>
                            <a className="nav-link active" href="#resume" 
                            onClick={() => { setAboutSelected(false); setContactSelected(false); setWorkSelected(false); setResumeSelected(true); }}>
                            <h3>Resume</h3></a>
                        </li>
                    </ul>
            </div>
        </nav>
        </header>

    )
};

export default Nav;