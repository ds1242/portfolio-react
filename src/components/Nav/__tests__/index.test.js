import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '../Nav';


afterEach(cleanup);

describe('Nav component', () => {
    it('renders', () => {
        render(<Nav />);
    });
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav />);

        expect(asFragment()).toMatchSnapshot();
    });
});

describe('links are visible', () => {
    it('inserts text into the links', () => {
        const { getByTestId } = render(<Nav />);

        expect(getByTestId('link')).toHaveTextContent(`David's Portfolio`);
        expect(getByTestId('about')).toHaveTextContent('About Me');
    })
})