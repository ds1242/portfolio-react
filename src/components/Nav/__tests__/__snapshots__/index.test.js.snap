// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component matches snapshot 1`] = `
<DocumentFragment>
  <header>
    <nav
      class="navbar navbar-expand-sm"
      data-testid="navbar"
    >
      <div
        class="container-fluid "
      >
        <a
          class="navbar-brand"
          data-testid="link"
          href="/"
        >
          <h1>
            David's Portfolio
          </h1>
        </a>
        <ul
          class="navbar-nav "
        >
          <li
            class="undefined"
          >
            <a
              class="nav-link active"
              data-testid="about"
              href="#about"
            >
              <h3>
                About Me
              </h3>
            </a>
          </li>
          <li
            class="undefined"
          >
            <a
              class="nav-link active"
              href="#work"
            >
              <h3>
                My Projects
              </h3>
            </a>
          </li>
          <li
            class="undefined"
          >
            <a
              class="nav-link active"
              href="#contact-me"
            >
              <h3>
                Contact
              </h3>
            </a>
          </li>
          <li
            class="undefined"
          >
            <a
              class="nav-link active"
              href="#resume"
            >
              <h3>
                Resume
              </h3>
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </header>
</DocumentFragment>
`;
